File: _CodeTopics/LeetCode_contest/weekly/weekly2021/231/231_1.py
class Solution(object):
    def checkOnesSegment(self, s):
        """
        :type s: str
        :rtype: bool
        """
        
        flagZero = False
        for i in range(1, len(s)):
            if not flagZero and s[i] == '0':
                flagZero = True
            if flagZero and s[i] == '1':
                return False
        return True
    
"""
https://leetcode-cn.com/submissions/detail/151989679/

191 / 191 个通过测试用例
状态：通过
执行用时: 28 ms
内存消耗: 12.9 MB
"""

File: _CodeTopics/LeetCode/1601-1800/001784--weekly231_1/tran/231_1.cpp
class Solution {
public:
    bool checkOnesSegment(string s) {

        // 第 231 场周赛第一题
        // https://leetcode-cn.com/submissions/detail/151989679/

        bool flagZero = false;
        for (int i = 1; i < s.size(); ++i) {
            if (!flagZero && s[i] == '0') {
                flagZero = true;
            }
            if (flagZero && s[i] == '1') {
                return false;
            }
        }
        return true;
    }
};

/*
https://leetcode.cn/submissions/detail/369397150/

执行用时：
4 ms
, 在所有 C++ 提交中击败了
17.44%
的用户
内存消耗：
6.1 MB
, 在所有 C++ 提交中击败了
18.22%
的用户
通过测试用例：
191 / 191
*/
